### PR TITLE
Create Creating AWS Volume Snapshots in Kubernetes Using the EBS CSI Driver

### PR DESCRIPTION
**Pull request by golkir**
I'd like to pitch this article for the K8s blog. Please, review

### CODE REVIEW

File: content/en/blog/_posts/Creating AWS Volume Snapshots in Kubernetes Using the EBS CSI Driver [hunk 1]
```diff
@@ -0,0 +1,651 @@
...
+
+To complete these examples, you'll also need:
+
+- an AWS account with the ability to create IAM users
+- AWS CLI
+- kubectl
+
+In what follows, we'll first quickly provision a K8s cluster on AWS using Kops. The cluster will be configured with all required feature gates.
+
+**NOTE**: all code snippets and manifests used in this tutorial can be obtained from this [GitHub repo](https://github.com/supergiant/blog-code-examples/tree/master/volumesnapshots-aws-ebs-csi).
```
Comment (sftim) [inline content/en/blog/_posts/Creating AWS Volume Snapshots in Kubernetes Using the EBS CSI Driver]: You should be able to use the Hugo syntax for a note in blog entries, I think.
(I haven't tried). That'll give a consistent appearance.

File: content/en/blog/_posts/Creating AWS Volume Snapshots in Kubernetes Using the EBS CSI Driver [hunk 2]
```diff
@@ -0,0 +1,651 @@
...
+```
+
+Let's verify that our data is still there:
+
+```shell
+export MYSQL_POD=$(kubectl get pods -l app=mysql-clone --no-headers | awk '{print $1}')
+
+kubectl exec -ti $MYSQL_POD -- bash
+mysql --user=root --password=password
+show databases;
```
Comment (sftim) [inline content/en/blog/_posts/Creating AWS Volume Snapshots in Kubernetes Using the EBS CSI Driver]: This is nside a code block marked as POSIX shell script. Maybe it works better to not declare a language?

The main documentation's style guide uses one code block for the command and a second one for the expected output.

File: content/en/blog/_posts/Creating AWS Volume Snapshots in Kubernetes Using the EBS CSI Driver [hunk 3]
```diff
@@ -0,0 +1,651 @@
...
+      - name: mysql-persistent-storage
+        persistentVolumeClaim:
+          claimName: pvc-restore
+```
+
+Save this spec to the mysql-clone.yaml , and create new Deployment:
+
+```shell
+kubectl create -f mysql-clone.yaml
+deployment.apps/mysql-clone created
```
Comment (sftim) [inline content/en/blog/_posts/Creating AWS Volume Snapshots in Kubernetes Using the EBS CSI Driver]: The Kubernetes documentation style guide [uses](https://kubernetes.io/docs/contribute/style/style-guide/#separate-commands-from-output) one code block for the command and a second one for the expected output.

**Comment by golkir** [comment]
@sftim thanks for your review.
> I wonder if this would work better as a [Task](https://kubernetes.io/docs/tasks/) page than a blog post?

It's up to you but I think it works for a blog post because the specific CSI driver is demonstrated.

**Comment by mrbobbytables** [comment]
> > I wonder if this would work better as a [Task](https://kubernetes.io/docs/tasks/) page than a blog post?
> 
> It's up to you but I think it works for a blog post because the specific CSI driver is demonstrated.

I'd defer to @kbarnard10 It is specific, but it is much closer to a full blown tutorial or task.

**Comment by sftim** [comment]
@golkir - if you added a Task page around this idea, I think there's still enough material and comment that you can blog about it and link to the new Task page from the blog post.

How does that sound?

**Comment by castrojo** [comment]
I think summarizing/high leveling more in the blog post, and then having the technical content as a task could be a good idea and then linking to the task as the canonical version of the document going forward. A concern I have is people usually don't go back and update old blog posts to ensure the technical parts are up to date.

**Comment by kbarnard10** [comment]
+1 to @sftim and @castrojo suggestions

**Comment by vonguard** [comment]
I am wicked busy for the next 2 weeks, but afterwards, I can do a phone
interview with y'all and write up as a blog post. If anyone wants to pick a
time after June 16, I will schedule around your availability

**Comment by golkir** [comment]
> @golkir - if you added a Task page around this idea, I think there's still enough material and comment that you can blog about it and link to the new Task page from the blog post.
> 
> How does that sound?

@sftim Yeah, that's probably a good idea.

**Comment by golkir** [comment]
> @golkir - if you added a Task page around this idea, I think there's still enough material and comment that you can blog about it and link to the new Task page from the blog post.
> 
> How does that sound?

How should I go about it? Update the blog request (delete the "tutorial" part) and add more high-level description? And create a separate pull request for the Task?

**Comment by sftim** [comment]
@golkir, [2019-03-07-raw-block-volume-support-to-beta](https://github.com/kubernetes/website/blob/master/content/en/blog/_posts/2019-03-07-raw-block-volume-support-to-beta.md) is a decent starting point for an announcement-type article, and [2019-03-22-e2e-testing-for-everyone](https://github.com/kubernetes/website/blob/master/content/en/blog/_posts/2019-03-22-e2e-testing-for-everyone.md) is a fair example of a howto style article.

Overall I feel that blog articles work well if people can read through them in a few minutes, whereas Task pages are OK even if it takes much longer to work through. Because of that, I favor slightly shorter blog posts than have come through in the past.

As a _suggestion_: take the tutorial section from this PR and use that to and make a new PR that adds a brand new task page. Reword / rework _this_ PR to look more the examples I gave, including front matter, with a shorter set of steps to give readers a taste of the details. The reworked blog entry can then hyperlink to the task, for readers to follow if their appetite is whetted.

**Comment by golkir** [comment]
> @golkir, [2019-03-07-raw-block-volume-support-to-beta](https://github.com/kubernetes/website/blob/master/content/en/blog/_posts/2019-03-07-raw-block-volume-support-to-beta.md) is a decent starting point for an announcement-type article, and [2019-03-22-e2e-testing-for-everyone](https://github.com/kubernetes/website/blob/master/content/en/blog/_posts/2019-03-22-e2e-testing-for-everyone.md) is a fair example of a howto style article.
> 
> Overall I feel that blog articles work well if people can read through them in a few minutes, whereas Task pages are OK even if it takes much longer to work through. Because of that, I favor slightly shorter blog posts than have come through in the past.
> 
> As a _suggestion_: take the tutorial section from this PR and use that to and make a new PR that adds a brand new task page. Reword / rework _this_ PR to look more the examples I gave, including front matter, with a shorter set of steps to give readers a taste of the details. The reworked blog entry can then hyperlink to the task, for readers to follow if their appetite is whetted.

@sftim thanks! got it

**Comment by kbarnard10** [comment]
@golkir I'm going to close this PR for now. Please re-open with the updated version once you're ready. Reach out to us at #sig-docs-blog on the Kubernetes Slack if you have any questions!

**Pull request closed without merging**